engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/blueprint/NadelExecutionBlueprintFactory.kt

```diff
@@ -171,9 +184,11 @@ private class Factory(
                     return@mapNotNull null
                 }
 
-                val underlyingParentType = getUnderlyingType(hydratedFieldParentType)
-                    ?: error("No underlying type for: ${hydratedFieldParentType.name}")
-                val fieldDefs = underlyingParentType.getFieldsAlong(inputValueDef.valueSource.queryPathToField.segments)
+                val fieldDefs = mapFirstNotNullFromUnderlyingType(hydratedFieldParentType) { underlyingParentType ->
```

Comment:
Look at `mapFirstNotNullFromUnderlyingType`'s documentation.

So basically we're finding the first underlying type that has `getFieldsAlong(inputValueDef.valueSource.queryPathToField.segments)` as NOT null, and getting that value.